pylint/checkers/utils.py

```diff
@@ -203,6 +203,48 @@ def is_builtin(name):
     """
     return name in builtins or name in SPECIAL_BUILTINS
 
+
+def is_defined_in_scope(var_node, varname, scope):
```

Comment:
@sushobhit27 I think this refactoring needs to be done in a separate commit. Now it's not clear where the fix actually is, while if the function would have been moved in a separate commit, it would have been more obvious.

Reply:
@PCManticore actually fix is due to below code, rest is refactoring:
```
 if isinstance(scope, astroid.If):
        for node in scope.body:
            if ((isinstance(node, astroid.AssignName) and node.name == varname)
                    or (isinstance(node, astroid.Nonlocal) and varname in node.names)):
                return True
```
So I will split it into two commits, first containing only refactor code and second one having above additional code.